src/Bicep.Core/Configuration/ModuleAliasesConfiguration.cs

```diff
@@ -1,50 +1,142 @@
+    }
 
     public class ModuleAliasesConfiguration
     {
+        private static readonly Regex ModuleAliasNameRegex = new(@"^[\w-]+$");
+
         private ModuleAliasesConfiguration(
             ImmutableDictionary<string, TemplateSpecModuleAlias> templateSpecModuleAliases,
-            ImmutableDictionary<string, BicepRegistryModuleAlias> bicepRegistryModuleAliases)
+            ImmutableDictionary<string, OciArtifactModuleAlias> ociArtifactModuleAliases,
+            string resourceName)
         {
             this.TemplateSpecModuleAliases = templateSpecModuleAliases;
-            this.BicepRegistryModuleAliases = bicepRegistryModuleAliases;
+            this.OciArtifactModuleAliases = ociArtifactModuleAliases;
+            this.ResourceName = resourceName;
         }
 
-        public static ModuleAliasesConfiguration Bind(IConfiguration rawConfiguration)
+        public static ModuleAliasesConfiguration Bind(IConfiguration rawConfiguration, string resourceName)
         {
             var templateSpecModuleAliases = new Dictionary<string, TemplateSpecModuleAlias>();
-            rawConfiguration.GetSection("ts").Bind(templateSpecModuleAliases);
+            rawConfiguration.GetSection(ModuleReferenceSchemes.TemplateSpecs).Bind(templateSpecModuleAliases);
 
-            var bicepRegistryModuleAliases = new Dictionary<string, BicepRegistryModuleAlias>();
-            rawConfiguration.GetSection("br").Bind(bicepRegistryModuleAliases);
+            var bicepRegistryModuleAliases = new Dictionary<string, OciArtifactModuleAlias>();
+            rawConfiguration.GetSection(ModuleReferenceSchemes.Oci).Bind(bicepRegistryModuleAliases);
 
-            return new(templateSpecModuleAliases.ToImmutableDictionary(), bicepRegistryModuleAliases.ToImmutableDictionary());
+            return new(templateSpecModuleAliases.ToImmutableDictionary(), bicepRegistryModuleAliases.ToImmutableDictionary(), resourceName);
         }
 
+        // TODO: do not expose the two dictionaries as properties.
         public ImmutableDictionary<string, TemplateSpecModuleAlias> TemplateSpecModuleAliases { get; }
 
-        public ImmutableDictionary<string, BicepRegistryModuleAlias> BicepRegistryModuleAliases { get; }
+        public ImmutableDictionary<string, OciArtifactModuleAlias> OciArtifactModuleAliases { get; }
+
+        /// <summary>
+        /// Gets the built-in configuraiton manifest resource name if the configuration is a built-in configuraiton,
+        /// or a path to a bicepconfig.json file if the configuration is a custom one.
+        /// </summary>
+        public string ResourceName { get; }
+
+        public ModuleAlias? TryGetModuleAlias(string scheme, string aliasName, out ErrorBuilderDelegate? errorBuilder)
+        {
+            if (!ModuleAliasNameRegex.IsMatch(aliasName))
+            {
+                errorBuilder = x => x.InvalidModuleAliasName(aliasName);
+                return null;
+            }
+
+            return scheme switch
```

Comment:
>switch [](http://example.com/codeflow?start=26&length=6)

This logic is very similar to what `IModuleRegistryProvider` does. Is it possible to depend on it (with modifications)?

Reply:
Done refactoring. This method was removed as it's not necessary.